### DISCUSSION TITLE
add radio browser feature

### DISCUSSION THREAD

**Discussion by Greylinux** [jcorporation/myMPD#492] [Ideas]
Hi, first of all after many hours of research , I found your project and it is by far the best looking and the simplest to install and use  with my MPD server. ( seriously a lot of pain and late nights have been endured so far  :smile: ) so thankyou so much for all your efforts in maintaining what seems to be a declining number of MPD clients.

anyway to the the request , do you think it would be possible to add a radio browser tab that utilises  the api from [radio-browser]( https://www.radio-browser.info/#/) 
that way you could search for radio streams rather than add singularly 

thanks again for efforts 

kind regards

**Comment by jcorporation** [comment]
Many thanks for your feedback. Staring myMPD helps to improve the visibility of the project.

I checked shortly the radio-browser homepage. The service looks greate. I will only implement api clients for opensource/-data services.

Let us discuss how the service should be integrated.

**Reply by Greylinux** [reply]
Hi, I couldn't agree more , as far as I know the API is opensource 
[API](https://api.radio-browser.info/) 

Is this what your were referring to ?

**Reply by jcorporation** [reply]
Yes, the api is open and the data is public domain. Thats greate!

**Reply by Greylinux** [reply]
I think a few mpd client already implemeny this feature but in my opinion are overly bloated and complicated to use , if there were a radio browser tab on the sidebar that provided a search option that used this API then users could add which radio stations they wanted rather than have a load already preset .

**Reply by Greylinux** [reply]
This is one of [them](https://github.com/krisek/audioloader/)  , advertised in the app section of radio-browser.
I don't know how they implement it as I could never get the web client to work , the install would always fail and visually I'm not a huge fan of there design

**Reply by Greylinux** [reply]
Having a quick look at that mpd client , it looks like they import a [python client ](https://github.com/andreztz/pyradios) application that incorporates radio browser for them.

**Reply by Greylinux** [reply]
This nodejs  [module](https://github.com/nepodev/radio-browser) might  be a better fit though

**Reply by Greylinux** [reply]
And another one [here](https://github.com/piffio/node-tunein-radio)  also

**Reply by jcorporation** [reply]
The myMPD backend is written in C. I think I must write my own client api implementation.

**Reply by Greylinux** [reply]
ah! so it will be quite a task to implement then?

**Reply by jcorporation** [reply]
It would be a bigger task. myMPD has no backend structure for remote api calls. I must design this before implementing  a specific api. But it is in the roadmap for listenbrainz/musicbrainz integration.

**Reply by Greylinux** [reply]
Maybe a temporary solution would be to have a radio tab on the sidebar that allows you to store streams added manually, rather than pinning streams added as playlist to the home screen which is the only current option.

**Reply by jcorporation** [reply]
I do not want to invest time in a temporary solution.

**Comment by jcorporation** [comment]
In the radiobrowser branch there is a first implementation of the radiobrowse feature. Actually you can search the stations by name, country and tags and it can be handled like other streams (added to homescreen, playlists, queue, etc).

What other functions should be implemented for this feature?

Implementation is tracked in issue #502

**Comment by Emilot** [comment]
JC amazing job(as always) and v9.0.0 is a major enchancement.

I do not know if you can make something about it or if it is possible.

If a webradio station has metadata (like Radio Paradise), covers, etc...can you display them somehow?

Thank you

**Comment by jcorporation** [comment]
Metadata display from the radio-browser api is planed.

**Comment by jcorporation** [comment]
I recently merged the radiobrowser feature into the v9.1.0 release branch. Now it is a good time for initial testing this new feature.

**Reply by matthiakl** [reply]
I'm really looking forward to this feature, so I tried out the v9.1.0 branch. While searching the WebradioDB works great, I get an "Illegal response" popup when searching radio-browser.info. Any ideas?

**Reply by jcorporation** [reply]
Commit dc83561bdd569e1478efd0ad0bf4e28421170d8c fixes a bug related to this feature.

**Reply by matthiakl** [reply]
It works now, thanks.

**Comment by jcorporation** [comment]
I used the radio-browser.info api now a little bit. It seems the data quality is poor. I am unsure if this integration has a big benefit.

The problem seems to be the disabled editing feature:
https://github.com/segler-alex/radiobrowser-api-rust/issues/49

I thinking about a moderated list of radio stations, like moode delivers.

Any thoughts on this?

**Reply by jcorporation** [reply]
First attempt to create webradio files compatible with myMPD: https://github.com/jcorporation/webradiodb
Most of the webradios are automatically converted from the moode audio webradios list.

Additional webradios can be added by opening an issue.

It would be great if this becomes a community driven collection of webradios.

**Comment by jcorporation** [comment]
I merged the v9.1.0 branch into devel. All features for the new version are implemented and the stabilization phase begins.

**Reply by opengk** [reply]
Hi, this looks like a great step and a very exciting feature. I'm using mympd since a few months now and I love it. You did a great work, thanks a lot for developing this project.

Indeed it would be tremendous to participate and propose new radios. I might consider installing the devel branch to test it.

**Reply by jcorporation** [reply]
Thanks!

You can also add webradios without installing myMPD: https://jcorporation.github.io/webradiodb/

**Reply by onkobu** [reply]
I also really appreciate it. Before there were only playlists without any meta data. Now there are pictures, URLs and a description. Thanks a lot.
